esphome/components/mitsubishi/mitsubishi.cpp

```diff
@@ -22,37 +40,213 @@ const uint16_t MITSUBISHI_HEADER_MARK = 3500;
+    traits.add_supported_mode(climate::CLIMATE_MODE_FAN_ONLY);
+
+  // Default to only 3 levels in ESPHome even if most unit supports 4. The 3rd level is not used.
+  traits.set_supported_fan_modes(
+      {climate::CLIMATE_FAN_AUTO, climate::CLIMATE_FAN_LOW, climate::CLIMATE_FAN_MEDIUM, climate::CLIMATE_FAN_HIGH});
+  if (this->fan_mode_ == MITSUBISHI_FAN_Q4L)
+    traits.add_supported_fan_mode(climate::CLIMATE_FAN_QUIET);
+  if (/*this->fan_mode_ == MITSUBISHI_FAN_5L ||*/ this->fan_mode_ >= MITSUBISHI_FAN_4L)
+    traits.add_supported_fan_mode(climate::CLIMATE_FAN_MIDDLE);  // Shouldn't be used for this but it helps
+
+  traits.set_supported_swing_modes({climate::CLIMATE_SWING_OFF, climate::CLIMATE_SWING_BOTH,
+                                    climate::CLIMATE_SWING_VERTICAL, climate::CLIMATE_SWING_HORIZONTAL});
+
+  traits.set_supported_presets(
+      {climate::CLIMATE_PRESET_NONE, climate::CLIMATE_PRESET_ECO, climate::CLIMATE_PRESET_BOOST});
+
+  return traits;
+}
+
 void MitsubishiClimate::transmit_state() {
-  uint32_t remote_state[18] = {0x23, 0xCB, 0x26, 0x01, 0x00, 0x20, 0x08, 0x00, 0x30,
-                               0x58, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00};
+  // Byte 0-4: Constant: 0x23, 0xCB, 0x26, 0x01, 0x00
+  // Byte 5: On=0x20, Off: 0x00
+  // Byte 6: MODE (See MODEs above (Heat/Dry/Cool/Auto/FanOnly)
+  // Byte 7: TEMP bits 0,1,2,3, added to MITSUBISHI_TEMP_MIN
+  //          Example: 0x00 = 0°C+MITSUBISHI_TEMP_MIN = 16°C; 0x07 = 7°C+MITSUBISHI_TEMP_MIN = 23°C
+  // Byte 8: MODE_A & Wide Vane (if present)
+  //          MODE_A bits 0,1,2 different than Byte 6 (See MODE_As above)
+  //          Wide Vane bits 4,5,6,7 (Middle = 0x30)
+  // Byte 9: FAN/Vertical Vane/Switch To Auto
+  //          FAN (Speed) bits 0,1,2
+  //          Vertical Vane bits 3,4,5 (Auto = 0x00)
+  //          Switch To Auto bits 6,7
+  // Byte 10: CLOCK Current time as configured on remote (0x00=Not used)
+  // Byte 11: END CLOCK Stop time of HVAC (0x00 for no setting)
+  // Byte 12: START CLOCK Start time of HVAC (0x00 for no setting)
+  // Byte 13: Constant 0x00
+  // Byte 14: HVAC specfic, i.e. ECONO COOL, CLEAN MODE, always 0x00
+  // Byte 15: HVAC specfic, i.e. POWERFUL, SMART SET, PLASMA, always 0x00
+  // Byte 16: Constant 0x00
+  // Byte 17: Checksum: SUM[Byte0...Byte16]
+
+  uint32_t remote_state[18] = {0x23, 0xCB, 0x26, 0x01, 0x00, 0x20, 0x08, 0x00, 0x00,
+                               0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00};
 
   switch (this->mode) {
-    case climate::CLIMATE_MODE_COOL:
-      remote_state[6] = MITSUBISHI_COOL;
-      break;
     case climate::CLIMATE_MODE_HEAT:
-      remote_state[6] = MITSUBISHI_HEAT;
+      remote_state[6] = MITSUBISHI_MODE_HEAT;
+      remote_state[8] = MITSUBISHI_MODE_A_HEAT;
+      break;
+    case climate::CLIMATE_MODE_DRY:
+      if (this->supports_dry_) {
+        remote_state[5] = MITSUBISHI_OFF;  // Better safe than sorry
+        break;
+      }
```

Comment:
```suggestion
      if (this->supports_dry_) {
        remote_state[5] = MITSUBISHI_OFF;  // Better safe than sorry
        break;
      }
```
I think this needs to be removed. CLIMATE_MODE_DRY is only loaded if supports_dry is set to True. In this code if someone tries to set the unit to dry, i.e CLIMATE_MODE_DRY, meaning supports_dry is true the unit is turned OFF.